### ISSUE TITLE
Consider adding a C# linter github workflow

### ISSUE THREAD

**Issue by quisquous** [OverlayPlugin/OverlayPlugin#11]
It'd be nice to have C# linting taken care of automatically so that folks don't have to care about this when looking at PRs.

**Comment by valarnin** [comment]
I'm thinking about tackling this after #82 is merged.

Is there a particular style that the project should focus on, e.g. defaults for [StyleCop](https://github.com/DotNetAnalyzers/StyleCopAnalyzers)? Should I just pick a file (e.g. `OverlayPlugin.Common/Registry.cs`) and enforce the closest to that style across the project?

**Comment by quisquous** [comment]
I'd suggest starting small and adding one rule (e.g. line endings) and the associated workflow (and vscode?) hooks.  Past that, I think conforming to existing code is a good option.  Alternatively (or in addition to?), following a reasonable style guide like "Microsoft C# style guide" sort of thing could make sense here too.

Personally I don't have a lot of strong feelings on style, other than wanting it automated as much as possible.  I am not a C# programmer usually so I don't have a lot of sense of what is "normal" for projects.

**Comment by xpdota** [comment]
Same, I don't really care as long as it's something that can be auto-formatted.

**Comment by valarnin** [comment]
This was solved by #111
